app.js

```diff
@@ -17,17 +17,29 @@ app.get('/api/articles/:article_id', getArticleByArticleId)
 
 app.get('/api/articles/:article_id/comments', getCommentsByArticleId )
 
+app.post('/api/articles/:article_id/comments', postComment)
+
 app.use((err, request, response, next) => {
-  if (err.status === 404) {
+  if (err.status === 404 || err.code === '23503') {
     response.status(404).send({ msg: 'No article found!' });
   } else {
     next(err);
   }
 });
 
 app.use((err, request, response, next) => {
-  if (err.code === '22P02') {
-    response.status(400).send({ msg: 'Bad request, no id found!' });
+  if (err.code === '22P02'||err.code === '23502'){
+    response.status(400).send({ msg: 'Bad Request!' });
+  // } if (){
+  //   response.status(400).send({ msg: 'Bad Request!'})
+  }else{
+    next(err)
+  }
+})
+
+app.use((err, req, res, next) => {
+  if(err.status && err.msg) {
+    res.status(err.status).send({msg: err.msg})
   }else{
     next(err)
   }
```

Comment:
Well handled errors! Might be worth considering putting these functions in their own file then requiring them in